base/bucket_gocb.go

```diff
@@ -1651,10 +1662,31 @@ func (bucket *CouchbaseBucketGoCB) WriteUpdateWithXattr(k string, xattrKey strin
 				value = nil
 				xattrValue = nil
 			}
+
+			// If user xattrs is enabled
+			if userXattrKey != "" {
+				userXattrCas, err := bucket.GetXattr(k, userXattrKey, &userXattrValue)
+				switch pkgerrors.Cause(err) {
+				case gocb.ErrKeyNotFound, gocb.ErrSubDocBadMulti, nil:
+					// Occurs if doc / xattr is unavailable or no error
```

Comment:
Treating success (nil) the same as not found here makes the error check on line 1680 hard to reason about. I think it would be better to define a separate nil case in the switch, and do the cas match check in there.